File: LeetCode/String/guessWord.cpp
#include <iostream>
#include <vector>
#include <string>
#include<stdio.h>
using namespace std;

class Master{
    public :
        void guessWord(){

        }
};


class Solution {
public:
    void findSecretWord(vector<string>& words, string word) {
        int length = words.size();
        int n = 6;
        cout << "length " << length << endl;

        int allowedGuess = 10;
        Master.guess(eiowzz);
        //MinMax Approach
        

        for(int i = 0; i < length; i++){
            int matches = 0;
            for(int j = 0; j < n; j++){
                if(words[i][j] == word[j]){
                    matches++;
                }
            }
            if(matches == n){
                cout << "The guess word found at position " << i << endl;
            }
        }
    }
};



int main(){
    vector<string>str = {"acckzz","ccbazz","eiowzz","abcczz"};
    string word = "eiowzz";
    Solution s;
    s.findSecretWord(str, word);
    return 0;
}

File: LeetCode/String/python/guess_word.py
wordlist = ["acckzz","ccbazz","eiowzz","abcczz"]

def matches(a: str, b: str) -> int:
    matchCount = 0
    for i in range(len(a)):
        if a[i] == b[i]:
            matchCount += 1
    return matchCount

for _ in range(10) :
    # initialize as an empty dict (not a set)
    count: dict[str, int] = {}
    # safe access with get
    for w1 in wordlist :
        for w2 in wordlist :
            if matches(w1,w2) == 0:
                print(f"w1 and w2 {w1,w2}")
                continue
            count[w1] = count.get(w1,0) + 1